data_processors/pipeline/services/metadata_srv.py

```
@@ -195,3 +196,65 @@ def get_metadata_for_library_runs(library_runs: List[LibraryRun]) -> List[LabMet
     for lib_run in library_runs:
         library_id_list.append(lib_run.library_id)
     return get_metadata_by_keywords_in(libraries=library_id_list)
+
+
+def get_sorted_library_id_by_sequencing_time(library_ids: List[str]) -> str:
+    """
+    Business logic:
+    For given library_id list, you'd like to get which library_id to use for triggering a workflow.
+    Using latest-greatest strategy -- such that
+
+        Check through LibraryRun > SequenceRun(PendingAnalysis)
+        Order library by SequenceRun date_modified descending
+        Return single library_id string; latest by their sequencing time
+
+    Example use case, see https://github.com/umccr/data-portal-apis/issues/547
+
+    :return: str library_id (latest by their sequencing time)  (OR) empty string on any fault condition and upto caller
+    to evaluate on returning empty string
```

Throw an exception? 
If we pass valid library IDs we'd normally expect a return, no?

---

That's the alternate solution, yes. We can raise runtime exception. That would crash it; instead of silent skip. Do we prefer? I am ok either way. Pretty much metadata <> run issue at this point; so, raising it sound good too.

---

Hm... yes if there's a metadata <> run issue it should probably end in an exception. This is on the metadata serving side...
We can still catch it in the consumer if we want to ignore it...

---

Ok; I will raise it.

---

It now raises with https://github.com/umccr/data-portal-apis/commit/a22f1969532aa7eb6a9dd8e7f06cd9d5506495c7